src/panels/lovelace/entity-rows/hui-media-player-entity-row.ts

```diff
@@ -68,47 +94,131 @@ class HuiMediaPlayerEntityRow extends LitElement implements LovelaceRow {
+                    ? "hass:volume-off"
+                    : "hass:volume-high"}
+                  @click=${this._toggleMute}
+                ></paper-icon-button>
+              `
+            : ""}
+          ${supportsFeature(stateObj, SUPPORT_VOLUME_SET) && !this._narrow
+            ? html`
+                <ha-slider
+                  .dir=${computeRTLDirection(this.hass!)}
+                  .value=${Number(stateObj.attributes.volume_level) * 100}
+                  pin
+                  @change=${this._selectedValueChanged}
+                  ignore-bar-touch
+                  id="input"
+                ></ha-slider>
+              `
+            : supportsFeature(stateObj, SUPPORT_VOLUME_BUTTONS) &&
+              !this._veryNarrow
+            ? html`
+                <paper-icon-button
+                  icon="hass:volume-minus"
+                  @click=${this._volumeDown}
+                ></paper-icon-button>
+                <paper-icon-button
+                  icon="hass:volume-plus"
+                  @click=${this._volumeUp}
+                ></paper-icon-button>
+              `
+            : ""}
+        </div>
+        <div class="controls">
+          ${supportsFeature(stateObj, SUPPORT_PREVIOUS_TRACK) &&
+          !this._veryNarrow
+            ? html`
+                <paper-icon-button
+                  icon="hass:skip-previous"
+                  @click=${this._previousTrack}
+                ></paper-icon-button>
+              `
+            : ""}
+          ${stateObj.state !== "playing" &&
+          !supportsFeature(stateObj, SUPPORTS_PLAY)
+            ? ""
+            : html`
+                <paper-icon-button
+                  icon=${this._computeControlIcon(stateObj)}
+                  @click=${this._playPause}
+                ></paper-icon-button>
+              `}
+          ${supportsFeature(stateObj, SUPPORT_NEXT_TRACK)
+            ? html`
+                <paper-icon-button
+                  icon="hass:skip-next"
+                  @click=${this._nextTrack}
+                ></paper-icon-button>
+              `
+            : ""}
+        </div>
+      </div>
     `;
   }
 
   static get styles(): CSSResult {
     return css`
+      :host {
+        display: block;
+      }
+      .flex {
+        display: flex;
+        align-items: center;
+        padding-left: 48px;
+        justify-content: space-between;
+      }
+      .volume {
+        display: flex;
+        flex-grow: 2;
+        flex-shrink: 2;
+      }
       .controls {
         white-space: nowrap;
       }
+      ha-slider {
+        flex-grow: 2;
+        flex-shrink: 2;
+        width: 100%;
+      }
     `;
   }
 
+  private _attachObserver(): void {
+    if (typeof ResizeObserver !== "function") {
+      import("resize-observer").then((modules) => {
+        modules.install();
+        this._attachObserver();
+      });
+      return;
+    }
+
+    this._resizeObserver = new ResizeObserver(() =>
```

Comment:
We should clean this up in `disconnectedCallback` and re-attach in `connectedCallback`

Reply:
like this?
```ts
public connectedCallback(): void {
    this._attachObserver();
  }

  public disconnectedCallback(): void {
    this._resizeObserver = undefined;
  }
```

Reply:
You may need to unobserve first

Reply:
trying to attach in connectedCallback did not work. Added unobserve to the disconnectedCallback.

Reply:
So now it is not observed when you switch a tab and go back, why didnt the observe on connected work?

Reply:
Not sure. Nothing loaded at all then with no errors

Reply:
@iantrich 
```js
public connectedCallback(): void {
  super.connectedCallback(); // This is the reason you weren't getting anything loaded
  if (!this._resizeObserver) {
      this._attachObserver();
  }
}
```

Reply:
omg...